### PR TITLE
Change StateDB for HashDB and use FlushID to improve sequencer's performance

### LINKED ISSUE #2067: Addapt node to new HashDB and Executor Protos
https://github.com/0xPolygonHermez/zkevm-comms-protocol/pull/76/files

### LINKED ISSUE #2098: Use Flush ID to stored processed transactions in the sequencer once they are confirmed into the HashDB

### LINKED ISSUE #2178: Testing and improving integration for the new approach with HashDB using flush IDs

### PR DESCRIPTION
**Pull request by ToniRamirezM**
Closes #2067
Closes #2098
Closes #2178 

### What does this PR do?

- Changes StateDB for HashDB
- Uses flush ID to store processed transactions

### Reviewers

Main reviewers:

- @ARR552 
- @agnusmor 
- @tclemos

### CODE REVIEW

**Review by ARR552 — APPROVED** [review]

File: sequencer/finalizer.go [hunk 1]
```diff
@@ -273,11 +300,84 @@ func (f *finalizer) halt(ctx context.Context, err error) {
...
+	}
+}
+
 // newWIPBatch closes the current batch and opens a new one, potentially processing forced batches between the batch is closed and the resulting new empty batch
 func (f *finalizer) newWIPBatch(ctx context.Context) (*WipBatch, error) {
 	f.sharedResourcesMux.Lock()
 	defer f.sharedResourcesMux.Unlock()
 
+	// Wait until all processed transactions are saved
+	for len(f.pendingTransactionsToStore) > 0 {
```
Comment (Psykepro) [inline sequencer/finalizer.go:376]: Using a `WaitGroup` for waiting is way better than sleeping. As it could take less than 100 ms.

**Review by Psykepro — CHANGES_REQUESTED** [review]

**Review by Psykepro — APPROVED** [review]

File: sequencer/finalizer.go [hunk 2]
```diff
@@ -20,7 +20,9 @@ import (
 	"github.com/jackc/pgx/v4"
 )
 
-const oneHundred = 100
+const (
+	oneHundred = 100
```
Comment (agnusmor) [inline sequencer/finalizer.go:24]: I think this constant is not used. In the storePendingTransactions we are using the "100" value directly (with  //nolint:gomnd). Maybe we need to replace the "100" for this oneHundred const (or delete this const as is not used)

File: sequencer/finalizer.go [hunk 3]
```diff
@@ -273,11 +296,86 @@ func (f *finalizer) halt(ctx context.Context, err error) {
...
 func (f *finalizer) newWIPBatch(ctx context.Context) (*WipBatch, error) {
 	f.sharedResourcesMux.Lock()
 	defer f.sharedResourcesMux.Unlock()
 
+	// Wait until all processed transactions are saved
+	startWait := time.Now()
+	f.pendingTransactionsToStoreWG.Wait()
+	endWait := time.Now()
+
+	log.Info("waiting for pending transactions to be stored took: ", endWait.Sub(startWait).String())
```
Comment (agnusmor) [inline sequencer/finalizer.go:377]: it will be good to put in the message log the unit of the time (miliseconds(ms)?)

**Comment by ToniRamirezM** [comment]
This changes are included in #2196

**Pull request closed without merging**